### ISSUE TITLE
Symbol Layers should support stretchable images with a fixed middle, not just fixed edges

### ISSUE THREAD

**Issue by avi-c** [mapbox/mapbox-gl-native-ios#569]
GL supports stretchable symbol images that can have fixed Left, Right, Top, and Bottom edges and stretch the middle portion. There are also important cases where it's useful to have a stretchable image where the middle is fixed and the edges stretch. See the images 

**Platform:** iOS
**Mapbox SDK version:** v6.0

Today we can form images such as:
![Left](https://user-images.githubusercontent.com/10037825/101438414-3132eb00-38c7-11eb-8a7c-b1a2b1dd79fc.png)
![Right](https://user-images.githubusercontent.com/10037825/101438416-31cb8180-38c7-11eb-9fc6-72dad59997ca.png)

I would like to make a stretchable symbol image like this:

![Middle](https://user-images.githubusercontent.com/10037825/101438452-414aca80-38c7-11eb-860c-1a764ed30bb8.png)

**Comment by avi-c** [comment]
My particular use case is for labeling intersecting roads during navigation, but the feature would be generally useful too:

![Intersections Dark 2 - Small](https://user-images.githubusercontent.com/10037825/102270483-e4f13780-3ed2-11eb-882b-e1e0a64ca833.gif)

**Comment by mkrussel77** [comment]
This can be accomplished on both Android and web.  It seems very common to want a pointer in the center of the text background pointing to the location on the map the text is referring.

Its even the example for Mapbox-GLJS.  https://docs.mapbox.com/mapbox-gl-js/example/add-image-stretchable/

**Comment by avi-c** [comment]
After talking to @ClareTrainor I realized that this support does exist on GL. Seems like the issue is either something missing in the iOS SDK or just me not knowing that it is there. Either way it's not a GL issue so I will close the issue.

**Comment by mkrussel77** [comment]
The GL portion does support this, but seems like the ticket should remain open until a public API to use the GL support has been added for iOS.

**Comment by avi-c** [comment]
Yeah, the iOS SDK doesn't expose the needed support for centered pins. Leaving this open for further comment or fixing.
